### PR TITLE
Chore: remove `UseState` + add support for controls in `TimeOfDayPicker`/`TimeRangeInput` story

### LINKED ISSUE #53039: Add controls to `TimeOfDayPicker` story

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/53039

**Special notes for your reviewer**:

### CODE REVIEW

**Review by JoaoSilvaGrafana — COMMENTED** [review]
This looks great! I've got two points only, the first one is an extra ask 😬 
1) I had no idea what `isReversed` or `hideQuickRanges` did, could we add props documentation for the two components if it is not too much work?
2) When `clearable` is true on `TimeRangeInput` and I do clear it I get an "Invalid time value" error on storybook, which did not happen previously

**Comment by ashharrison90** [comment]
@JoaoSilvaGrafana all updated 👍

File: .betterer.results [hunk 1]
```diff
@@ -1397,11 +1397,6 @@ exports[`better eslint`] = {
     "packages/grafana-ui/src/components/DateTimePickers/TimeOfDayPicker.tsx:5381": [
       [0, 0, 0, "Unexpected any. Specify a different type.", "0"]
     ],
-    "packages/grafana-ui/src/components/DateTimePickers/TimeRangeInput.tsx:5381": [
-      [0, 0, 0, "Unexpected any. Specify a different type.", "0"],
-      [0, 0, 0, "Do not use any type assertions.", "1"],
-      [0, 0, 0, "Do not use any type assertions.", "2"]
-    ],
```
Comment (JoaoSilvaGrafana) [inline .betterer.results:1404]: 🎉

**Review by JoaoSilvaGrafana — APPROVED** [review]
Nice!

**Pull request merged**